metrics-operator/converter/slo_converter.go

```diff
@@ -0,0 +1,259 @@
+	target := &metricsapi.Target{}
+	// clean up % criteria
+	o = cleanupObjective(o)
+	// skip objective target conversion if it has criteria combined with logical OR -> not supported
+	// this way the SLO will become "informative"
+	if shouldIgnoreObjective(o) {
+		return target, nil
+	}
+
+	// if warning criteria are not defined, negotiate the pass criteria to create fail criteria
+	if len(o.Warning) == 0 {
+		if len(o.Pass) > 0 {
+			if len(o.Pass[0].Operators) > 0 {
+				// TODO cover use cases with multiple operators (create new objectives)
+				op, err := setupOperator(o.Pass[0].Operators[0])
+				if err != nil {
+					return target, err
+				}
+				target.Failure = op
+				return target, nil
+			}
+		}
+	}
+
+	// if warning criteria are defined, create new criteria with the following logic:
+	// !(warn criteria) -> fail criteria
+	// !(pass criteria) -> warn criteria
+	var err error
+	if len(o.Pass) > 0 {
+		if len(o.Pass[0].Operators) > 0 {
+			// TODO cover use cases with multiple operators (create new objectives)
+			op, err := setupOperator(o.Pass[0].Operators[0])
+			if err != nil {
+				return target, err
+			}
+			target.Warning = op
+		}
+		if len(o.Warning[0].Operators) > 0 {
+			// TODO cover use cases with multiple operators (create new objectives)
+			op, err := setupOperator(o.Warning[0].Operators[0])
+			if err != nil {
+				return target, err
+			}
+			target.Failure = op
+		}
+	}
+
+	return target, err
+}
+
+func cleanupObjective(o *Objective) *Objective {
+	o.Pass = cleanupCriteria(o.Pass)
+	o.Warning = cleanupCriteria(o.Warning)
+	return o
+}
+
+// remove % operators from criterium structure
+// if criterium did have only % operators, remove it from strucutre
+func cleanupCriteria(criteria []Criteria) []Criteria {
+	newCriteria := make([]Criteria, 0, len(criteria))
+	for _, c := range criteria {
+		operators := make([]string, 0, len(c.Operators))
+		for _, op := range c.Operators {
+			// keep only criteria with real values, not percentage
+			if !strings.Contains(op, "%") {
+				operators = append(operators, op)
+			}
+		}
+		// if criterium does have operator, store it
+		if len(operators) > 0 {
+			newCriteria = append(newCriteria, Criteria{Operators: operators})
+		}
+	}
+
+	return newCriteria
+}
+
+func shouldIgnoreObjective(o *Objective) bool {
+	return len(o.Pass) > 1 || len(o.Warning) > 1
+}
+
+// create operator for Target
+func setupOperator(op string) (*metricsapi.Operator, error) {
+	// remove whitespaces
+	op = strings.Replace(op, " ", "", -1)
+
+	operators := []string{"<=", "<", ">=", ">"}
+	for _, operator := range operators {
+		if strings.HasPrefix(op, operator) {
+			return createOperator(operator, strings.TrimPrefix(op, operator))
+		}
+	}
+
+	return &metricsapi.Operator{}, fmt.Errorf("invalid operator: '%s'", op)
+}
+
+// checks and negotiates the existing operator
+func createOperator(op string, value string) (*metricsapi.Operator, error) {
+	v, err := strconv.ParseInt(value, 10, 64)
```

Comment:
Let's go with float here bcs I think I could set criteria like the following in Keptn v1: `< 3.14`